source/MaterialXGenShader/ShaderGraph.cpp

```diff
@@ -1128,6 +1224,29 @@ void ShaderGraph::populateInputColorTransformMap(ColorManagementSystemPtr colorM
     }
 }
 
+void ShaderGraph::populateInputUnitTransformMap(UnitSystemPtr unitSystem, ShaderNodePtr shaderNode, ValueElementPtr input, const string& targetUnitSpace, const string& unitType)
+{
+    ShaderInput* shaderInput = shaderNode->getInput(input->getName());
+    const string& sourceUnitSpace = input->getUnit();
+    if (shaderInput && !sourceUnitSpace.empty())
+    {
+        if (shaderInput->getType() != Type::COLOR2 &&
```

Comment:
May be better to create a whitelist vs a blacklist, since this would allow float arrays, strings, matrices etc to go through.